src/commandfilter/mod.rs

```diff
@@ -0,0 +1,48 @@
+use crate::raw;
+use crate::RedisString;
+use std::os::raw::c_int;
+use std::ptr;
+
+pub struct CommandFilterContext {
+    pub ctx: *mut raw::RedisModuleCommandFilterCtx,
+}
+
+impl CommandFilterContext {
+    pub fn new(ctx: *mut raw::RedisModuleCommandFilterCtx) -> Self {
+        CommandFilterContext { ctx }
+    }
+
+    pub fn args_count(&self) -> usize {
+        unsafe { raw::RedisModule_CommandFilterArgsCount.unwrap()(self.ctx) as usize }
+    }
+
+    pub fn args_get(&self, pos: usize) -> RedisString {
+        let arg = unsafe {
+            raw::RedisModule_CommandFilterArgGet.unwrap()(self.ctx, pos as c_int)
+                as *mut raw::RedisModuleString
+        };
+        RedisString::new(ptr::null_mut(), arg)
+    }
+
+    pub fn args_insert(&self, pos: usize, arg: RedisString) -> usize {
+        // retain arg to since RedisModule_CommandFilterArgInsert going to release it too
+        raw::string_retain_string(std::ptr::null_mut(), arg.inner);
```

Comment:
Is it certain that `arg.inner` would be released by redis?
(I do not see it directly done in `RM_CommandFilterArgInsert`)

In comment about `RM_CommandFilterArgInsert`
I see:
```
The specified RedisModuleString argument may be used by Redis
 * after the filter context is destroyed, so it must not be auto-memory
 * allocated, freed or used elsewhere.
```

And also please see comment about `args_replace` when `pos` is out of bounds.

Reply:
yes they will be released like every other arguments